### PR TITLE
#116 - Conversion to interval matrix

### LINKED ISSUE #116: Conversion to interval matrix
```julia
julia> convert(Interval, 1.0)
[1, 1]

julia> convert(IntervalMatrix, A) # not defined

julia> Aint = IntervalMatrix([Interval(x) for x in A]) # current workaround
```
Otherwise, we could add a constructor `IntervalMatrix(A::Matrix{Float64})`, compare to `Interval(1)`.

### PR DESCRIPTION
**Pull request by mforets**
Closes #116.

### CODE REVIEW

File: src/matrix.jl [hunk 1]
```diff
@@ -95,6 +95,21 @@ function similar(M::IntervalMatrix)
...
 
+# constructor from a scalar matrix
+function IntervalMatrix(A::AbstractMatrix{T}) where {T<:Number}
+    return IntervalMatrix(map(Interval, A))
+end
+
+# conversion from scalar matrix to interval matrix
+function convert(::Type{<:IntervalMatrix}, A::AbstractMatrix{T}) where {T<:Number}
+    return IntervalMatrix(map(x -> convert(Interval, x), A))
+end
```
Comment (schillic) [inline src/matrix.jl]: I would just use the first function in the second function.
```julia
function convert(::Type{<:IntervalMatrix}, A::AbstractMatrix{T}) where {T<:Number}
    return IntervalMatrix(A)
end
```

File: src/matrix.jl [hunk 2]
```diff
@@ -95,6 +95,21 @@ function similar(M::IntervalMatrix)
...
+
+# no-op
+function convert(::Type{<:IntervalMatrix}, A::IntervalMatrix)
+    return A
+end
```
Comment (schillic) [inline src/matrix.jl]: This is not correct for `convert(IntervalMatrix{T2}, A::IntervalMatrix{T1})` with `T1 != T2`?

File: test/runtests.jl [hunk 3]
```diff
@@ -17,6 +17,10 @@ end
     @test m2 isa IntervalMatrix{Float64} && size(m2) == (2, 2)
     m3 = similar(m1)
     @test m3 isa IntervalMatrix{Float64} && size(m3) == size(m1)
+    m = [1.0 2.0; 3.0 4.0]
+    mint = [Interval(1) Interval(2); Interval(3) Interval(4)]
```
Comment (schillic) [inline test/runtests.jl]:
```suggestion
    mint = IntervalMatrix([Interval(1) Interval(2); Interval(3) Interval(4)])
```
I am actually surprised that the tests work.

File: src/matrix.jl (hunk 2 above)
Comment (mforets) [reply src/matrix.jl]: True; for the general case we should appropriate type promotion.

File: src/matrix.jl (hunk 1 above)
Comment (mforets) [reply src/matrix.jl]: But then this is not performing a conversion, right?

```julia
julia> x = Interval(1f0)
[1, 1]

julia> typeof(x)
Interval{Float32}


julia> convert(Interval{Float64}, x)
[1, 1]

julia> typeof(ans)
Interval{Float64}

Comment (mforets) [reply src/matrix.jl]: Then we wouldn't be doing any conversion; the intended behavior was to map the scalar case,

```julia
julia> x = Interval(1f0)
[1, 1]

julia> typeof(x)
Interval{Float32}

julia> convert(Interval{Float64}, x)
[1, 1]

julia> typeof(ans)
Interval{Float64}
```

Comment (schillic) [reply src/matrix.jl]: Okay, then there should be the faster `convert` for cases where the types agree.

**Comment by mforets** [comment]
I have some doubts about the conversion, so i have split the scalar constructor to https://github.com/JuliaReach/IntervalMatrices.jl/pull/135.

**Pull request closed without merging**